packages/core/core-flows/src/promotion/workflows/update-promotions-status.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  AdditionalData,
+  PromotionStatusValues,
+} from "@medusajs/framework/types"
+import { MedusaError, PromotionStatus } from "@medusajs/framework/utils"
+import {
+  WorkflowResponse,
+  createHook,
+  createStep,
+  createWorkflow,
+} from "@medusajs/framework/workflows-sdk"
+import { updatePromotionsStep } from "../steps"
+
+export type UpdatePromotionsStatusWorkflowInput = {
+  promotionsData: {
+    id: string
+    status: PromotionStatusValues
+  }[]
+} & AdditionalData
+
+export const updatePromotionsValidationStep = createStep(
+  "update-promotions-validation",
+  async function ({ promotionsData }: UpdatePromotionsStatusWorkflowInput) {
+    for (const promotionData of promotionsData) {
+      const allowedStatuses: PromotionStatusValues[] =
+        Object.values(PromotionStatus)
+
+      if (!allowedStatuses.includes(promotionData.status)) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `promotion's status should be one of - ${allowedStatuses.join(", ")}`
+        )
+      }
+    }
+  }
+)
+
+export const updatePromotionsStatusWorkflowId = "update-promotions-status"
+export const updatePromotionsStatusWorkflow = createWorkflow(
+  updatePromotionsStatusWorkflowId,
+  (input: UpdatePromotionsStatusWorkflowInput) => {
+    updatePromotionsValidationStep(input)
+
+    const updatedPromotions = updatePromotionsStep(input.promotionsData)
+
+    const promotionStatusUpdated = createHook("promotionStatusUpdated", {
+      promotions: updatedPromotions,
+      additional_data: input.additional_data,
+    })
```

Comment:
q: Why was this added? The `updatePromotionsWorkflow` already has a hook.

Reply:
It was a comment from Seb on the RFC where he wanted a workflow that you can hook into when status changes

Reply:
Ah alright, I hadn't seen. Would have thought this could be dealt with by the existing hook, but I am fine keeping it as is.

Reply:
Same, this workflow doesn't add much overhead, so just created one.